### PR TITLE
Feature suggestions see issue #610

### PR DESCRIPTION
**Pull request by chrisrapson**
This branch includes 3 commits addressing suggestions numbers 3,1,4 in Issue #610. I think these ones improve the tool without any downside. Suggestions 2,5,6 (and unnumbered 7) I think are improvements, but either there is a tradeoff or they are more subjective, so I would prefer to discuss first, before making a pull request.

https://github.com/tzutalin/labelImg/issues/610

### CODE REVIEW

**Comment by tzutalin** [comment]
Looks good to me. But it will be better to put displayed text to string res.

**Comment by chrisrapson** [comment]
Sorry, I don't understand what you mean. Do you mean the text that is displayed in the MessageBox? The string whose variable name at the moment is `msg`? Happy to make adjustments, just trying to clarify so that I make the right ones.

File: labelImg.py [hunk 1]
```diff
@@ -1392,12 +1392,22 @@ def resetAll(self):
...
+                return True
+            else:
+                return False
 
     def discardChangesDialog(self):
-        yes, no = QMessageBox.Yes, QMessageBox.No
-        msg = u'You have unsaved changes, proceed anyway?'
-        return yes == QMessageBox.warning(self, u'Attention', msg, yes | no)
+        yes, no, cancel = QMessageBox.Yes, QMessageBox.No, QMessageBox.Cancel
+        msg = u'You have unsaved changes, would you like to save them and proceed?\nClick "No" to undo all changes.'
```
Comment (tzutalin) [inline labelImg.py:1409]: Cloud we move this message to https://github.com/tzutalin/labelImg/blob/master/resources/strings/strings.properties as we will support other languages?

File: libs/canvas.py [hunk 2]
```diff
@@ -224,13 +234,18 @@ def mousePressEvent(self, ev):
             if self.drawing():
                 self.handleDrawing(pos)
             else:
-                self.selectShapePoint(pos)
+                selection = self.selectShapePoint(pos)
                 self.prevPoint = pos
-                self.repaint()
```
Comment (tzutalin) [inline libs/canvas.py:229]: may I know what is the reason to remove repaint?

**Review by tzutalin — APPROVED** [review]

**Comment by chrisrapson** [comment]
it looks like you've already done the merge. Sorry I wasn't able to reply earlier.

Yes, good idea to move the string to strings.properties for internationalisation.

Going from memory (it is a month of so since I made these changes) I think there was another call of `repaint()` in the function which called `mousePressEvent` and I didn't think it needed to be called twice. Either that, or I read somewhere that another Qt function (`show()`? `redraw()`?) was more efficient that `repaint()`). In any case, I didn't detect any problems in testing. If you're interested, I can look in more detail to try and figure out what my thought process was.

**Pull request merged**